### PR TITLE
MT-v2: Prompt studio core variable replacement minor changes from PR 611

### PR DESCRIPTION
**Pull request by chandrasekharan-zipstack**
## What

- v2 changes for https://github.com/Zipstack/unstract/pull/611/files
- This includes one of the [files from 600](https://github.com/Zipstack/unstract/pull/600/files#diff-2e0e822d13fd3b1d3eec03559262ce23884bb318ace82f7fcf6a777ca20541ff) as well, but its unused in this PR

## Why

- As part of multi tenancy v2 migration

## Can this PR break any existing features. If yes, please list possible items. If no, please explain why. (PS: Admins do not merge the PR without this section filled)

- No, this code is unused

## Notes on Testing

- No explicit testing done

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: backend/prompt_studio/prompt_studio_core_v2/exceptions.py [hunk 1]
```diff
@@ -67,3 +67,22 @@ class MaxProfilesReachedError(APIException):
         f"Maximum number of profiles (max {ProfileManagerKeys.MAX_PROFILE_COUNT})"
         " per prompt studio project has been reached."
     )
+
+
+class OperationNotSupported(APIException):
+    status_code = 403
```
Comment (harini-venkataraman) [inline backend/prompt_studio/prompt_studio_core_v2/exceptions.py:73]: @chandrasekharan-zipstack Handled this as a part of the app. https://github.com/Zipstack/unstract/pull/720

File: backend/prompt_studio/prompt_studio_core_v2/prompt_variable_service.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import re
+from enum import Enum
+from typing import Any
+
+from prompt_studio.prompt_studio.models import ToolStudioPrompt
```
Comment (muhammad-ali-e) [inline backend/prompt_studio/prompt_studio_core_v2/prompt_variable_service.py:5]: should import from V2

File: backend/prompt_studio/prompt_studio_core_v2/exceptions.py (hunk 1 above)
Comment (chandrasekharan-zipstack) [reply backend/prompt_studio/prompt_studio_core_v2/exceptions.py:73]: We'll take care of it if there are any conflicts

File: backend/prompt_studio/prompt_studio_core_v2/prompt_variable_service.py (hunk 2 above)
Comment (muhammad-ali-e) [reply backend/prompt_studio/prompt_studio_core_v2/prompt_variable_service.py]: addressed

**Review by muhammad-ali-e — APPROVED** [review]
LGTM, We can address conflict once the latest PRs came in

**Review by hari-kuriakose — APPROVED** [review]
@chandrasekharan-zipstack LGTM overall.

Suggest to add dev comments containing examples of what the regexes will ultimately match, as that really helps.

**Comment by hari-kuriakose** [comment]
Duplication will be handled when we remove v1 later.

**Pull request merged**